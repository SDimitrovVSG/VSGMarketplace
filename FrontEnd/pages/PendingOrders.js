import { pendingOrdersApp } from "../src/pendingOrdersApp.js";

const PendingOrders = () => {
    const main = document.querySelector('main');
    main.id = "pendingOrdersMain";
    main.innerHTML =  `
    <div id="headingSection">
        <span class="codeColumn">Code</span>
        <span class="qtyColumn">QTY</span>
        <span class="priceColumn">Price</span>
        <span class="orderedByColumn">Ordered By</span>
        <span class="orderDateColumn">Order Date</span>
        <span class="actionColumn">Action</span>
    </div>
`;

    pendingOrdersApp();
};

export default PendingOrders;
