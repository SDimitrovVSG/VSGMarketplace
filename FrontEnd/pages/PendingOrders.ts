import { pendingOrderComponent } from "../components/pendingOrderComponent.js";
import { loadPendingOrders } from "../src/itemsService.js";

const PendingOrders = async (): Promise<void> => {
    const main = document.querySelector('main') as HTMLElement;
    main.id = "pendingOrdersMain";
    main.innerHTML = `
    <div id="headingSection">
        <span class="codeColumn">Code</span>
        <span class="qtyColumn">QTY</span>
        <span class="priceColumn">Price</span>
        <span class="orderedByColumn">Ordered By</span>
        <span class="orderDateColumn">Order Date</span>
        <span class="actionColumn">Action</span>
    </div>
    `;

    type Order = {
        id: number;
        code: String;
        quantity: number;
        price: number;
        email: string
        date: string;
        status: string;
    };
    const orders: Order[] = await loadPendingOrders();
    orders.forEach((p: Order) => {
        pendingOrderComponent(p);
    });
};

export default PendingOrders;
