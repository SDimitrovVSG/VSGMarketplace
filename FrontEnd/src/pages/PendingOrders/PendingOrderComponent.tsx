import { useCompleteOrderMutation } from '../../services/ordersService';
import { IOrder } from '../../types/types';
import { useRef } from 'react';
import { toast } from 'react-toastify';
import { Fade } from '@mui/material';

type PendingOrderProps = {
    order: IOrder;
}

const PendingOrderComponent = ({ order }: PendingOrderProps) => {
    const [completeOrder] = useCompleteOrderMutation();
    const orderRef = useRef<HTMLDivElement>(null);

    const onComplete = async () => {
        const response = await completeOrder(order.id);
        if ('error' in response) {
            return;
        }

        orderRef.current?.remove();
        toast.success('Order completed!');
    }

    return (
        <Fade in={true} timeout={1000}>
            <div ref={orderRef} className='order' id={order.id.toString()} role='cell'>
                <div className='firstThree'>
                    <span className='codeColumn'>{order.productCode}</span>
                    <span className='qtyColumn'>{order.quantity}</span>
                    <span className='priceColumn'>{order.price} BGN</span>
                </div>
                <span className='orderedByColumn'>{order.email}</span>
                <span className='orderDateColumn'>{order.date}</span>
                <button className='actionColumn completeButton' onClick={onComplete}>Complete</button>
            </div >
        </Fade>
    );
};

export default PendingOrderComponent;