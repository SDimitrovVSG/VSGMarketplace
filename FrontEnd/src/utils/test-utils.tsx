import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

type ProvidersProps = {
  children: ReactElement;
};

const AllTheProviders = ({ children }: ProvidersProps) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  )
}

const customRender = (ui: ReactElement, options?: object) => (
  render(ui, { wrapper: AllTheProviders, ...options })
);

export * from '@testing-library/react';

// override render method
export { customRender as render }