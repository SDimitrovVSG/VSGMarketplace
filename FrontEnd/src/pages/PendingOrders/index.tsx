import { useGetPendingOrdersQuery } from '../../services/ordersService.ts';
import { IOrder } from "../../types/types.ts";
import PendingOrderComponent from "./PendingOrderComponent.tsx";

const PendingOrders = () => {
    const { data: orders, isLoading } = useGetPendingOrdersQuery();

    return (
        <main id='pendingOrdersMain'>
            <div id="headingSection">
                <span className="codeColumn">Code</span>
                <span className="qtyColumn">QTY</span>
                <span className="priceColumn">Price</span>
                <span className="orderedByColumn">Ordered By</span>
                <span className="orderDateColumn">Order Date</span>
                <span className="actionColumn">Action</span>
            </div>
            {orders && orders.map((order: IOrder) => <PendingOrderComponent order={order} key={order.id} />)}
            {isLoading && <div className="order">Loading...</div>}
            {orders?.length === 0 && <div className="order">No orders</div>}
        </main>
    );
};

export default PendingOrders;
