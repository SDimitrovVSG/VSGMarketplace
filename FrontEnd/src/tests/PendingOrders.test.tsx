import { render, screen } from 'test-utils';
import mockServer from 'mock-server';
import PendingOrders from '../pages/PendingOrders';

describe('PendingOrders', () => {
    beforeAll(() => mockServer.listen());

    afterEach(() => mockServer.resetHandlers());

    afterAll(() => mockServer.close());
    it('should render with orders', async () => {
        render(<PendingOrders />);
        const orders = await screen.findAllByRole('cell');
        expect(orders).toHaveLength(2);
    });
    it('should match the snapshot', async () => {
        const { container } = render(<PendingOrders />);
        expect(container).toMatchSnapshot();
    });
});