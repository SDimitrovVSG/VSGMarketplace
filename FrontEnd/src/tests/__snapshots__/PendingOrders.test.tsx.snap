// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingOrders should match the snapshot 1`] = `
<div>
  <main
    id="pendingOrdersMain"
  >
    <div
      id="headingSection"
    >
      <span
        class="codeColumn"
      >
        Code
      </span>
      <span
        class="qtyColumn"
      >
        QTY
      </span>
      <span
        class="priceColumn"
      >
        Price
      </span>
      <span
        class="orderedByColumn"
      >
        Ordered By
      </span>
      <span
        class="orderDateColumn"
      >
        Order Date
      </span>
      <span
        class="actionColumn"
      >
        Action
      </span>
    </div>
    <div
      class="order"
      role="cell"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="firstThree"
      >
        <span
          class="codeColumn"
        >
          1
        </span>
        <span
          class="qtyColumn"
        >
          1
        </span>
        <span
          class="priceColumn"
        >
          12
           BGN
        </span>
      </div>
      <span
        class="orderedByColumn"
      >
        ERedzhepov@vsgbg.com
      </span>
      <span
        class="orderDateColumn"
      >
        2023-05-31 04:16
      </span>
      <button
        class="actionColumn completeButton"
      >
        Complete
      </button>
    </div>
    <div
      class="order"
      role="cell"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="firstThree"
      >
        <span
          class="codeColumn"
        >
          2
        </span>
        <span
          class="qtyColumn"
        >
          2
        </span>
        <span
          class="priceColumn"
        >
          12
           BGN
        </span>
      </div>
      <span
        class="orderedByColumn"
      >
        ERedzhepov@vsgbg.com
      </span>
      <span
        class="orderDateColumn"
      >
        2023-05-31 04:17
      </span>
      <button
        class="actionColumn completeButton"
      >
        Complete
      </button>
    </div>
  </main>
</div>
`;
